src/DynamoCore/Logging/DynamoLogger.cs

```diff
@@ -157,7 +157,7 @@ public IEnumerable<NotificationMessage> StartupNotifications
         /// </summary>
         /// <param name="debugSettings">Debug settings</param>
         /// <param name="logDirectory">Directory path where log file will be written</param>
-        public DynamoLogger(DebugSettings debugSettings, string logDirectory)
+        public DynamoLogger(DebugSettings debugSettings, string logDirectory, Boolean IsTestMode = false)
```

Comment:
Should we used chained constructors instead? Optional parameters require a client recompile, so technically this is a breaking change.

Reply:
NIT: The parameter name should start with lower case: `isTestMode`

Reply:
I finally found an official version of this:
https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/breaking-change-rules.md

but It does not seem as comprehensive as:
https://stackoverflow.com/questions/1456785/a-definitive-guide-to-api-breaking-changes-in-net

please review it whenever making changes to public apis.
relevant section:
`Adding a parameter with a default value.`
though it requires careful reading.

Reply:
Yes will be changing this to an overloaded constructor. I tried to look if adding a optional parameter is a good design or not for public API's but that link is really comprehensive